client/payment-details/dispute-details/dispute-steps.tsx

```diff
@@ -161,35 +152,10 @@ const DisputeSteps: React.FC< Props > = ( {
 									) }
 								/>
 							),
-							disputeduedate: (
-								<span className="dispute-steps__steps__response-date">
-									{ respondByDate }
-									<span
-										className={ classNames( {
-											'dispute-steps__steps__response-date--urgent':
-												daysRemaining < 3,
-											'dispute-steps__steps__response-date--warning':
-												daysRemaining < 7 &&
-												daysRemaining > 2,
-										} ) }
-									>
-										{ daysRemaining === 0
-											? __(
-													'(Last day today)',
-													'woocommerce-payments'
-											  )
-											: sprintf(
-													// Translators: %s is the number of days left to respond to the dispute.
-													_n(
-														'(%s day left to respond)',
-														'(%s days left to respond)',
-														daysRemaining,
-														'woocommerce-payments'
-													),
-													daysRemaining
-											  ) }
-									</span>
-								</span>
+							dueByDate: (
+								<DisputeDueByDate
+									dueBy={ dispute.evidence_details.due_by }
+								/>
```

Comment:
Love the way you've decoupled this into a self-contained component, and simplified props in the process.